src/metabase/util.clj

```diff
@@ -44,28 +45,64 @@
 
 (defprotocol ITimestampCoercible
   "Coerce object to a `java.sql.Timestamp`."
-  (->Timestamp ^java.sql.Timestamp [this]
+  (coerce-to-timestamp ^java.sql.Timestamp [this] [this timezone-coercible]
     "Coerce this object to a `java.sql.Timestamp`. Strings are parsed as ISO-8601."))
 
+(defprotocol ITimeZoneCoercible
+  "Coerce object to `java.util.TimeZone`"
+  (coerce-to-timezone ^TimeZone [this]
+    "Coerce `this` to `java.util.TimeZone`"))
+
+(extend-protocol ITimeZoneCoercible
+  String
+  (coerce-to-timezone [this]
+    (TimeZone/getTimeZone this))
+  TimeZone
+  (coerce-to-timezone [this]
+    this)
+  DateTimeZone
+  (coerce-to-timezone [this]
+    (.toTimeZone this)))
+
 (declare str->date-time)
 
+(def ^TimeZone utc
+  "UTC TimeZone"
+  (coerce-to-timezone "UTC"))
+
 (extend-protocol ITimestampCoercible
-  nil       (->Timestamp [_]
-              nil)
-  Timestamp (->Timestamp [this]
-              this)
-  Date       (->Timestamp [this]
-               (Timestamp. (.getTime this)))
+  nil
+  (coerce-to-timestamp [_]
+    nil)
+  Timestamp
+  (coerce-to-timestamp [this]
+    this)
+  Date
+  (coerce-to-timestamp
+    [this]
+    (coerce/to-timestamp (coerce/from-date this)))
   ;; Number is assumed to be a UNIX timezone in milliseconds (UTC)
-  Number    (->Timestamp [this]
-              (Timestamp. this))
-  Calendar  (->Timestamp [this]
-              (->Timestamp (.getTime this)))
-  ;; Strings are expected to be in ISO-8601 format. `YYYY-MM-DD` strings *are* valid ISO-8601 dates.
-  String    (->Timestamp [this]
-              (->Timestamp (str->date-time this)))
-  DateTime  (->Timestamp [this]
-              (->Timestamp (.getMillis this))))
+  Number
+  (coerce-to-timestamp [this]
+    (coerce/to-timestamp (coerce/from-long (long this))))
+  Calendar
+  (coerce-to-timestamp [this]
+    (coerce-to-timestamp (.getTime this)))
+  DateTime
+  (coerce-to-timestamp [this]
+    (coerce/to-timestamp this)))
+
+(defn ->Timestamp
```

Comment:
Oh I see 👀 

So does `coerce-to-timestamp` just power `->Timestamp`? Not intended for use elsewhere? We should make it `^:private` if so